software/dimos/src/shared/types.h

```diff
@@ -0,0 +1,8 @@
+#ifndef QT_DEMO_TYPES_H
+#define QT_DEMO_TYPES_H
```

Comment:
we use `#pragma once` in firmware instead of this, is that still supported with the compilers you're using now?